From 44bc6628ce7901cec394dab39a8f101bfe13f6f3 Mon Sep 17 00:00:00 2001
From: Jon Paul Maloy <jon.maloy@ericsson.com>
Date: Tue, 21 Jul 2015 06:42:28 -0400
Subject: [PATCH 061/376] tipc: fix compatibility bug

commit 16040894b26af9f85d9395f072c53d76a44eba21 upstream

In commit d999297c3dbbe7fdd832f7fa4ec84301e170b3e6
("tipc: reduce locking scope during packet reception") we introduced
a new function tipc_link_proto_rcv(). This function contains a bug,
so that it sometimes by error sends out a non-zero link priority value
in created protocol messages.

The bug may lead to an extra link reset at initial link establising
with older nodes. This will never happen more than once, whereafter
the link will work as intended.

We fix this bug in this commit.

Signed-off-by: Jon Maloy <jon.maloy@ericsson.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Zhenbo Gao <zhenbo.gao@windriver.com>
---
 net/tipc/link.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/net/tipc/link.c b/net/tipc/link.c
index 55b675d..b63d573 100644
--- a/net/tipc/link.c
+++ b/net/tipc/link.c
@@ -1639,7 +1639,7 @@ static int tipc_link_proto_rcv(struct tipc_link *l, struct sk_buff *skb,
 			rcvgap = peers_snd_nxt - l->rcv_nxt;
 		if (rcvgap || (msg_probe(hdr)))
 			tipc_link_build_proto_msg(l, STATE_MSG, 0, rcvgap,
-						  0, l->mtu, xmitq);
+						  0, 0, xmitq);
 		tipc_link_release_pkts(l, msg_ack(hdr));
 
 		/* If NACK, retransmit will now start at right position */
-- 
1.7.5.4

